flang/runtime/clock.cpp

```diff
@@ -43,7 +43,7 @@ void RTNAME(DateAndTime)(char *date, char *time, char *zone,
     copyBufferAndPad(date, dateChars, len);
   }
   if (time) {
-    auto ms{t.tv_usec / 1000};
+    long ms{t.tv_usec / 1000};
```

Comment:
The braced initialization might lead to warnings if `tv_usec` were ever declared with a type wider than `long`.  Consider using a cast of `t.tv_usec` to `long` within the braces instead.

Be advised, changes to the runtime in `fir-dev` might not persist.

Reply:
`auto` always avoids the declaration mismatch nonsense.

Why not fix the printf format specifier itself?

Reply:
> The braced initialization might lead to warnings if tv_usec were ever declared with a type wider than long.

tv_usec only has to hold values from 0 to 999,999 so I'm not worried about it having a wider type than long.

> auto always avoids the declaration mismatch nonsense.
> Why not fix the printf format specifier itself?

The printf specifier has to depend on the type of tv_usec: %03d or %03ld

Another alternative is `std::intmax_t ms{t.tv_usec / 1000};` and `%03jd`. Then there shouldn't be any narrowing problems.